pkg/lib/aws/aws.go

```diff
@@ -60,49 +51,65 @@ func NewFromClientS3Path(s3Path string, awsClient *Client) (*Client, error) {
 	return NewAnonymousClientWithRegion(region)
 }
 
-func NewFromEnvS3Path(s3Path string) (*Client, error) {
+func NewFromS3Path(s3Path string) (*Client, error) {
 	bucket, _, err := SplitS3Path(s3Path)
 	if err != nil {
 		return nil, err
 	}
-	return NewFromEnvS3Bucket(bucket)
+	return NewFromS3Bucket(bucket)
 }
 
-func NewFromEnvS3Bucket(bucket string) (*Client, error) {
+func NewFromS3Bucket(bucket string) (*Client, error) {
 	region, err := GetBucketRegion(bucket)
 	if err != nil {
 		return nil, err
 	}
-	return NewFromEnv(region)
+	return NewForRegion(region)
 }
 
-func NewFromCredsS3Path(s3Path string, accessKeyID string, secretAccessKey string) (*Client, error) {
-	bucket, _, err := SplitS3Path(s3Path)
+func NewForRegion(region string) (*Client, error) {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Region: aws.String(region),
+		},
+		SharedConfigState: session.SharedConfigEnable,
+	})
+
 	if err != nil {
-		return nil, err
+		return nil, errors.WithStack(err)
 	}
-	return NewFromCredsS3Bucket(bucket, accessKeyID, secretAccessKey)
-}
 
-func NewFromCredsS3Bucket(bucket string, accessKeyID string, secretAccessKey string) (*Client, error) {
-	region, err := GetBucketRegion(bucket)
+	if sess.Config.Credentials == nil {
+		return nil, ErrorUnableToFindCredentials()
+	}
+
+	_, err = sess.Config.Credentials.Get()
 	if err != nil {
-		return nil, err
+		return nil, ErrorUnableToFindCredentials()
 	}
```

Comment:
Check for access key ID and secret (and that it works with e.g. ec2 IAM roles) (also below)